### PR TITLE
add e2e test cases

### PR DESCRIPTION
**Pull request by syuhei176**
## What I did

* add 5 test scenarios
* fix finalizeExit
* fix event watcher
* fix getAllUserActions method

### CODE REVIEW

File: integration-test/e2e-tests/__tests__/index.test.ts [hunk 1]
```diff
@@ -4,41 +4,366 @@ import { LevelKeyValueStore } from '@cryptoeconomicslab/level-kvs'
...
-  beforeAll(async () => {
-    const kvs = new LevelKeyValueStore(Bytes.fromString('plasma_light_client'))
-    const wallet = new ethers.Wallet(
+  let aliceLightClient: LightClient
+  let bobLightClient: LightClient
+  let senderWallet: ethers.Wallet
+  let recieverWallet: ethers.Wallet
+
+  async function increaseBlock() {
+    for (let i = 0; i < 10; i++) {
```
Comment (tkmct) [inline integration-test/e2e-tests/__tests__/index.test.ts:37]: Does it increase block number without waiting?
And why would it need to send 10 transactions?

File: integration-test/e2e-tests/__tests__/index.test.ts [hunk 2]
```diff
@@ -4,41 +4,366 @@ import { LevelKeyValueStore } from '@cryptoeconomicslab/level-kvs'
...
+    await sleep(10000)
+
+    expect(await getBalance(aliceLightClient)).toEqual('0.05')
+
+    const exitList = await aliceLightClient.getExitList()
+    expect(exitList.length).toBe(1)
+
+    await increaseBlock()
+
+    await finalizeExit(aliceLightClient)
```
Comment (tkmct) [inline integration-test/e2e-tests/__tests__/index.test.ts:195]: Can we check postcondition of finalizeExit?

File: packages/eth-contract/src/contract/CommitmentContract.ts [hunk 3]
```diff
@@ -55,19 +55,26 @@ export class CommitmentContract implements ICommitmentContract {
...
+      await handler(
         BigNumber.fromString(blockNumber.toString()),
         FixedBytes.fromHexString(32, root)
       )
     })
-    this.eventWatcher.cancel()
-    this.eventWatcher.start(() => {
+  }
+
+  async startSubscribing() {
```
Comment (tkmct) [inline packages/eth-contract/src/contract/CommitmentContract.ts:70]: what about `startWatchingEvents` or something similar?

File: packages/plasma-light-client/src/LightClient.ts [hunk 4]
```diff
@@ -894,12 +924,12 @@ export default class LightClient {
...
    */
   public async getAllUserActions(): Promise<UserAction[]> {
-    const result: UserAction[] = []
+    let result: UserAction[] = []
     const currentBlockNumber = await this.commitmentContract.getCurrentBlock()
     let blockNumber = JSBI.BigInt(0)
     while (JSBI.lessThanOrEqual(blockNumber, currentBlockNumber.data)) {
       const actions = await this.getUserActions(BigNumber.from(blockNumber))
-      result.concat(actions)
+      result = result.concat(actions)
```
Comment (tkmct) [inline packages/plasma-light-client/src/LightClient.ts:932]: ooops, nice catch!

File: packages/eth-contract/src/contract/CommitmentContract.ts (hunk 3 above)
Comment (syuhei176) [reply packages/eth-contract/src/contract/CommitmentContract.ts:70]: seems good!

File: integration-test/e2e-tests/__tests__/index.test.ts (hunk 2 above)
Comment (syuhei176) [reply integration-test/e2e-tests/__tests__/index.test.ts:195]: ok, I'll check balance of PETH.

File: packages/eth-contract/src/contract/CommitmentContract.ts (hunk 3 above)
Comment (syuhei176) [reply packages/eth-contract/src/contract/CommitmentContract.ts]: changed

File: integration-test/e2e-tests/__tests__/index.test.ts (hunk 2 above)
Comment (syuhei176) [reply integration-test/e2e-tests/__tests__/index.test.ts:208]: added

File: packages/plasma-light-client/src/LightClient.ts (hunk 4 above)
Comment (syuhei176) [reply packages/plasma-light-client/src/LightClient.ts:932]: 👍

File: integration-test/e2e-tests/__tests__/index.test.ts (hunk 1 above)
Comment (syuhei176) [reply integration-test/e2e-tests/__tests__/index.test.ts:37]: > Does it increase block number without waiting?

When we doesn't specify blockTime option for ganache, ganache make new block instantly for every block.
https://github.com/trufflesuite/ganache-cli#options-1

Do you think we should use blockTime option and wait block creation?

> And why would it need to send 10 transactions?

There are no reasons😅

**Pull request merged**